components/threading/thread_viewer/thread_viewer.tsx

```diff
@@ -0,0 +1,564 @@
+            this.props.actions.getPostThread(this.props.selected.id);
+
+            if (
+                this.props.isCollapsedThreadsEnabled &&
+                this.props.userThread == null
+            ) {
+                this.fetchThread();
+            }
+        }
+
+        if (
+            this.props.isCollapsedThreadsEnabled &&
+            this.props.userThread?.id !== prevProps.userThread?.id
+        ) {
+            this.markThreadRead();
+        }
+
+        if (prevPostsArray.length >= curPostsArray.length) {
+            return;
+        }
+
+        const curLastPost = curPostsArray[0];
+
+        if (curLastPost.user_id === this.props.currentUserId || this.state.userScrolledToBottom) {
+            this.scrollToBottom();
+        }
+    }
+
+    public shouldComponentUpdate(nextProps: Props, nextState: State) {
+        if (!Utils.areObjectsEqual(nextState.postsArray, this.props.posts)) {
+            return true;
+        }
+
+        if (!Utils.areObjectsEqual(nextState.selected, this.props.selected)) {
+            return true;
+        }
+
+        if (!Utils.areObjectsEqual(nextProps.userThread, this.props.userThread)) {
+            return true;
+        }
+
+        if (nextProps.previewEnabled !== this.props.previewEnabled) {
+            return true;
+        }
+
+        if (nextState.isBusy !== this.state.isBusy) {
+            return true;
+        }
+
+        if (nextState.isScrolling !== this.state.isScrolling) {
+            return true;
+        }
+
+        if (nextState.topRhsPostId !== this.state.topRhsPostId) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private handleResize = (): void => {
+        this.setState({
+            windowWidth: Utils.windowWidth(),
+            windowHeight: Utils.windowHeight(),
+        });
+
+        if (UserAgent.isMobile() && document!.activeElement!.id === 'reply_textbox') {
+            this.scrollToBottom();
+        }
+        this.resizeRhsPostList();
+    }
+
+    private handleCardClick = (post: Post) => {
+        if (!post) {
+            return;
+        }
+
+        this.props.actions.selectPostCard(post);
+    }
+
+    private handleCardClickPost = (post: Post) => {
+        if (!post) {
+            return;
+        }
+
+        this.props.actions.selectPostCard(post);
+    }
+
+    private filterPosts = (posts: Post[], selected: Post | FakePost, openTime: number): Post[] => {
+        const postsArray: Post[] = [];
+
+        posts.forEach((cpost) => {
+            // Do not show empherals created before sidebar has been opened
+            if (cpost.type === 'system_ephemeral' && cpost.create_at < openTime) {
+                return;
+            }
+
+            if (cpost.root_id === selected.id) {
+                postsArray.unshift(cpost);
+            }
+        });
+
+        return postsArray;
+    }
+
+    public scrollToBottom = (): void => {
+        if ($('.post-right__scroll')[0]) {
```

Comment:
Can we avoid introducing a new JQuery dependency? We're aiming to drop JQuery in Q2 2021.

Reply:
(was moved/existing; didn't want to touch unnecessarily)